umbral/pre.py

```diff
@@ -129,15 +138,34 @@ def __bytes__(self):
         return self.to_bytes()
 
     def __eq__(self, other):
+        """
+        If both Capsules are activated, we compare only the activated components.
+        Otherwise, we compare only original components.
+        Each component is compared to its counterpart in constant time per the __eq__ of Point and BigNum.
+        """
         if all(self.activated_components() + other.activated_components()):
             activated_match = self.activated_components() == other.activated_components()
             return activated_match
         elif all(self.original_components() + other.original_components()):
             original_match = self.original_components() == other.original_components()
             return original_match
         else:
+            # This is not constant time obviously, but it's hard to imagine how this is valuable as
+            # an attacker already knows about her own Capsule.  It's possible that a Bob, having
+            # activated a Capsule, will make it available for comparison via an API amidst other
+            # (dormat) Capsules.  Then an attacker can, by alternating between activated and dormant
+            # Capsules, determine if a given Capsule is activated.  Do we care about this?
+            # Again, it's hard to imagine why.
             return False
 
+    def __hash__(self):
+        # We only ever want to store in a hash table based on original components;
+        # A Capsule that is part of a dict needs to continue to be lookup-able even
+        # after activation.
+        # Note: In case this isn't obvious, don't use this as a secure hash.  Use BLAKE2b or something.
+        component_bytes = tuple([component.to_bytes() for component in self.original_components()])
```

Comment:
I notice you are casting the a list comprehension to a tuple, is this because hash value is different if it were a generator expression?  Looks like you can drop the `[ ]`.